File: GFG/Mathematics/prime_factors.cpp
#include<iostream>
using namespace std;

bool isPrime(int q){
    if((q%2==0 && q!=2) or (q%3==0 && q!=3)){
        return false;
    }
    for(int i=5;i*i<=q;i++){
        if (q%i==0){
            return false;
        }
        i++;
    }
    return true;
}

void primefactors(int q){
    int i =2;
    cout<<"1 ";
    int qq=q;
    while (i<=qq){
        if (isPrime(i)){
            if (q%i==0){
                cout<<i<<" ";
                q=q/i;  
            }
            else{
                i++;
            }
        }
        else{
            i++;
        }
    }
}
int main(){
    int x;
    cin>>x;
    primefactors(x);
    return 0;

}

File: GFG/Mathematics/prime_factors.py
def isPrime(z):
    if (z!=2 and z%2==0):
        return False
    elif (z!=3 and z%3==0):
        return False
    i=5
    while (i*i<=z):
        if z%i==0:
            return False
        i+=1
    return True

def primefactors(q):
    i=2
    qq=q
    print("1",end=" ")
    while i<=qq:
        if isPrime(i):
            if q%i==0:
                print(f"{i}",end=" ")
                q=q//i
            else:
                i+=1
        else:
            i+=1
        #print(i,end=" ")
            
a=int(input("Enter number: "))
primefactors(a)
            
        
    